app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive: list["Animal"] = []
+
+    def __init__(self, name: str, helth: int = 100) -> None:
+        self.name = name
+        self.health = helth
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    @classmethod
+    def print_alive(cls) -> str:
+        return str([repr(animal) for animal in cls.alive])
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: "Herbivore") -> None:
```

Comment:
The type hint for the `herbivore` parameter is correct, but ensure that the `Herbivore` class is defined before this method is used, which it is in this case.